clientd3d/intrface.c

```diff
@@ -522,10 +474,17 @@ void PerformAction(int action, void *action_data)
       break;
 
    case A_TEXTCOMMAND:
-      TextInputSetText((char *) action_data, False);
-      ParseGotText((char *) action_data);
-      break;
-
+   {
+       DWORD now = timeGetTime();
```

Comment:
I'm not convinced this part is a good idea.

- It overrides the keyboard repeat rate from Windows system settings.
- Does this allow super fast macroing?